src/eventarc/eventarc.ts

```diff
@@ -0,0 +1,198 @@
+   * @returns Eventarc channel reference for publishing events.
+   */
+  public channel(name: string, options?: ChannelOptions): Channel;
+
+  /**
+   * Create a reference to the default Firebase channel:
+   * `locations/us-central1/channels/firebase`
+   *
+   * @param options - (optional) additional channel options
+   * @returns Eventarc channel reference for publishing events.
+   */
+  public channel(options?: ChannelOptions): Channel;
+ 
+  public channel(nameOrOptions?: string | ChannelOptions, options?: ChannelOptions): Channel {
+    let channel: string;
+    let opts: ChannelOptions;
+    if (validator.isNonEmptyString(nameOrOptions)) {
+      channel = nameOrOptions;
+    } else {
+      channel = 'locations/us-central1/channels/firebase';
+    }
+
+    if (validator.isNonNullObject(nameOrOptions)) {
+      opts = nameOrOptions as ChannelOptions;
+    } else {
+      opts = options as ChannelOptions;
+    }
+    let allowedEventTypes : string[] | undefined = undefined;
+    if (typeof opts?.allowedEventTypes === 'string') {
+      allowedEventTypes = opts.allowedEventTypes.split(',');
+    } else if (validator.isArray(opts?.allowedEventTypes)) {
+      allowedEventTypes = opts?.allowedEventTypes as string[];
+    } else if (typeof opts?.allowedEventTypes !== 'undefined') {
+      throw new FirebaseEventarcError(
+        'invalid-argument',
+        'AllowedEventTypes must be either an array of strings or a comma separated string.',
+      );
+    }
+    return new Channel(this, channel, allowedEventTypes);
+  }
+}
+
+/**
+ * Eventarc Channel.
+ */
+export class Channel {
+  private readonly eventarcInternal: Eventarc;
+  private nameInternal: string;
+
+  /**
+   * List if event types allowed by this channel for publishing. Other event types will be ignored.
+   */
+  public readonly allowedEventTypes?: string[]
+
+  private readonly client: EventarcApiClient;
+
+  /**
+   * @internal
+   */
+  constructor(eventarc: Eventarc, name: string, allowedEventTypes?: string[]) {
+    if (!validator.isNonNullObject(eventarc)) {
+      throw new FirebaseEventarcError(
+        'invalid-argument',
+        'First argument passed to Channel() must be a valid Eventarc service instance.',
+      );
+    }
+    if (!validator.isNonEmptyString(name)) {
+      throw new FirebaseEventarcError(
+        'invalid-argument', 'name is required.',
+      );
+    }
+
+    this.nameInternal = name;
+    this.eventarcInternal = eventarc;
+    this.allowedEventTypes = allowedEventTypes;
+    this.client = new EventarcApiClient(eventarc.app, this);
+  }
+
+  /**
+   * The {@link firebase-admin.eventarc#Eventarc} service instance associated with the current `Channel`.
+   *
+   * @example
+   * ```javascript
+   * var app = channel.eventarc;
+   * ```
+   */
+  get eventarc(): Eventarc {
+    return this.eventarcInternal;
+  }
+
+  /**
+   * The channel name as provided during channel creation except if it was not specifed then the default
```

Comment:
Suggest a stop, like "was not specified; then the..."

(note typo as well)

Reply:
Done.